### ISSUE TITLE
Error on Big Files

### ISSUE THREAD

**Issue by hectorbarranca** [ZonD80/mega-downloader#16]
Hi my friend.
If I tried to download big files (I am trying with a file of 1.3Gb) the file didn't download complety, when I saw the file it has about 300 to 500Mb.
Maybe if it possible to add partials download it can be fixed (as some gestor like JDownloader).
Not sure if we can do the partials download.
Thanks =).

**Comment by ZonD80** [comment]
Try to increase memory and time limits for script. Also use 64bit environment.

**Comment by hectorbarranca** [comment]
Thanks, I found the error that I have.

I check the code and I see that it send the header, in my code I send the headers too before the script so the script can't replace the headers.

I erased my headers and them it works perfect.

Also I found a MEGA archive that the mega_dec_attr didn't finish in "}" in the JSON, so I changed the function mega_dec_attr the line:
return json_decode(substr($attr, 4), true);

For the line:
return json_decode(substr($attr, 4, strrpos($attr,'}')-3), true);

Only for this type of archive (in the time that I used your script only happened with 1 archive), maybe it helps you.

Thanks again for the script.
